paddle/fluid/inference/CMakeLists.txt

```diff
@@ -50,47 +52,35 @@ if(NOT ON_INFER)
+              analysis_config paddle_pass_builder activation_functions ${mkldnn_quantizer_cfg})
+create_static_lib(paddle_fluid ${fluid_modules} ${STATIC_INFERENCE_API})
+if(NOT APPLE)
+  # TODO(liuyiqu: Temporarily disable the link flag because it is not support on Mac.
+  set(LINK_FLAGS "-Wl,--retain-symbols-file ${CMAKE_CURRENT_SOURCE_DIR}/paddle_fluid.sym")
+  set_target_properties(paddle_fluid PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
 endif()
 
-set(STATIC_INFERENCE_APIS paddle_fluid_api paddle_inference_api analysis_predictor)
-if (ANAKIN_FOUND)
+if(ANAKIN_FOUND)
     set(ANAKIN_SHARED_INFERENCE_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/api/api_anakin_engine.cc)
 endif()
 set(SHARED_INFERENCE_SRCS
-    io.cc ${CMAKE_CURRENT_SOURCE_DIR}/../framework/data_feed.cc ${CMAKE_CURRENT_SOURCE_DIR}/../framework/data_set.cc ${CMAKE_CURRENT_SOURCE_DIR}/../framework/data_feed_factory.cc ${CMAKE_CURRENT_SOURCE_DIR}/../framework/dataset_factory.cc ${CMAKE_CURRENT_SOURCE_DIR}/api/api.cc ${CMAKE_CURRENT_SOURCE_DIR}/api/api_impl.cc
+    io.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/../framework/data_feed.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/../framework/data_feed_factory.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/../framework/dataset_factory.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/api/api.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/api/api_impl.cc
     ${CMAKE_CURRENT_SOURCE_DIR}/api/analysis_predictor.cc
-    ${mkldnn_quantizer_src}
     ${CMAKE_CURRENT_SOURCE_DIR}/api/details/zero_copy_tensor.cc
+    ${mkldnn_quantizer_src_file}
     ${ANAKIN_SHARED_INFERENCE_SRCS})
 
-if(WIN32)
-  sep_library(paddle_fluid DEPS ${fluid_modules} ${STATIC_INFERENCE_APIS} zero_copy_tensor reset_tensor_array
-              analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-else(WIN32)
-  cc_library(paddle_fluid DEPS ${fluid_modules} ${STATIC_INFERENCE_APIS}
-            zero_copy_tensor reset_tensor_array analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-endif(WIN32)
-
-
-if(NOT APPLE)
-  # TODO(liuyiqu: Temporarily disable the link flag because it is not support on Mac.
-  set(LINK_FLAGS "-Wl,--retain-symbols-file ${CMAKE_CURRENT_SOURCE_DIR}/paddle_fluid.sym")
-  set_target_properties(paddle_fluid PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
-endif()
-
-# Create shared library
-if(WIN32)
-  sep_library(paddle_fluid_shared SHARED SRCS ${SHARED_INFERENCE_SRCS}
-              DEPS ${fluid_modules} paddle_fluid_api reset_tensor_array
-                  analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-else(WIN32)
-  cc_library(paddle_fluid_shared SHARED SRCS ${SHARED_INFERENCE_SRCS}
-            DEPS ${fluid_modules} paddle_fluid_api reset_tensor_array
-                  analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-endif()
+# Create shared inference library defaultly
+cc_library(paddle_fluid_shared SHARED SRCS ${SHARED_INFERENCE_SRCS}
+      DEPS ${fluid_modules} paddle_fluid_api analysis_predictor)
```

Comment:
check paddle_fluid_api is necessary or not

Reply:
It is not necessary, has been done.